R/filter_confirmation.R

```diff
@@ -0,0 +1,473 @@
+#'   by variables and the order.
+#'
+#'   *Default:* `"none"`
+#'
+#'   *Permitted Values:* `"none"`, `"warning"`, `"error"`
+#'
+#' @details
+#'
+#'   The following steps are performed to produce the output dataset.
+#'
+#'   ## Step 1
+#'
+#'   The input dataset is joined with itself by the variables specified for
+#'   `by_vars`. From the right hand side of the join only the variables
+#'   specified for `join_vars` are kept. The suffix ".join" is added to these
+#'   variables.
+#'
+#'   For example, for `by_vars = USUBJID`, `join_vars = vars(AVISITN, AVALC)` and input dataset
+#'
+#'   ```{r eval=FALSE}
+#'   # A tibble: 2 x 4
+#'   USUBJID AVISITN AVALC  AVAL
+#'   <chr>     <dbl> <chr> <dbl>
+#'   1             1 Y         1
+#'   1             2 N         0
+#'   ```
+#'
+#'   the joined dataset is
+#'
+#'   ```{r eval=FALSE}
+#'   A tibble: 4 x 6
+#'   USUBJID AVISITN AVALC  AVAL AVISITN.join AVALC.join
+#'   <chr>     <dbl> <chr> <dbl>        <dbl> <chr>
+#'   1             1 Y         1            1 Y
+#'   1             1 Y         1            2 N
+#'   1             2 N         0            1 Y
+#'   1             2 N         0            2 N
+#'   ```
+#'
+#'   ## Step 2
+#'
+#'   The joined dataset is restricted to observations where the joined variables
+#'   are at or after the other variables with respect to `order`.
+#'
+#'   The dataset from the example in the previous step with `order =
+#'   vars(AVISITN)` is restricted to
+#'
+#'   ```{r eval=FALSE}
+#'   A tibble: 4 x 6
+#'   USUBJID AVISITN AVALC  AVAL AVISITN.join AVALC.join
+#'   <chr>     <dbl> <chr> <dbl>        <dbl> <chr>
+#'   1             1 Y         1            1 Y
+#'   1             1 Y         1            2 N
+#'   1             2 N         0            2 N
+#'   ```
+#'
+#'   ## Step 3
+#'
+#'   If `first_cond` is specified, for each observation of the input dataset the
+#'   joined dataset is restricted to observations up to the first observation
+#'   where `first_cond` is fulfilled (the observation fulfilling the condition
+#'   is included). If for an observation of the input dataset the condition is
+#'   not fulfilled, the observation is removed.
+#'
+#'   ## Step 4
+#'
+#'   The joined dataset is grouped by the observations from the input dataset
+#'   and restricted to the observations fulfilling the condition specified by
+#'   `filter`.
+#'
+#'   ## Step 5
+#'
+#'   The first observation of each group is selected and the `*.join` variables
+#'   are dropped.
+#'
+#' @returns A subset of the observations of the input dataset. All variables of
+#'   the input dataset are included in the output dataset.
+#'
+#' @author Stefan Bundfuss
+#'
+#' @keywords adam user_utility
+#'
+#' @seealso [count_vals()], [min_cond()], [max_cond()]
+#'
+#' @export
+#'
+#' @examples
+#'
+#' library(tibble)
+#' library(admiral)
+#'
+#' # filter observations with a duration longer than 30 and
+#' # at, after, or up to 7 days before a COVID AE (ACOVFL == "Y")
```

Comment:
"at, after, or up to" -> "on or after" i think would read better

Reply:
Updated